python/elastic_search_serialization.py

```diff
@@ -0,0 +1,514 @@
+                        "pattern": "_",  # split all on underscore
+                        "replacement": " ",
+                    },
+                },
+                "filter": {
+                    "asciifolding_original": {
+                        "type": "asciifolding",
+                        "preserve_original": True,
+                    }
+                },
+                "analyzer": {
+                    # we define our custom analyzer
+                    "lk_analyzer": {
+                        "tokenizer": "standard",
+                        "char_filter": [
+                            "dot_to_whitespace",
+                            "underscore_to_whitespace",
+                        ],
+                        "filter": ["asciifolding_original", "lowercase", "stop"],
+                    }
+                },
+            },
+        }
+    },
+    "mappings": {
+        "dynamic_templates": [
+            {
+                # store a copy of labels/types in a non-analyzed way for filtering
+                "lk_type_has_raw": {  # for edges, the key is "lk_type_has_raw"
+                    "match": "lk_type",  # for edges, use "lk_type"
+                    "mapping": {
+                        "type": "text",
+                        "analyzer": "lk_analyzer",  # (default "lk_analyzer")
+                        "fields": {"raw": {"type": "keyword"}},
+                    },
+                }
+            },
+            {
+                "string": {
+                    "match": "*_lke_string",
+                    "mapping": {
+                        "type": "text",
+                        "analyzer": "lk_analyzer",  # default: "lk_analyzer"
+                    },
+                }
+            },
+            {"date": {"match": "*_lke_date", "mapping": {"type": "date"}}},
+            {"number": {"match": "*_lke_number", "mapping": {"type": "double"}}},
+            {"boolean": {"match": "*_lke_boolean", "mapping": {"type": "keyword"}}},
+        ]
+    },
+}
+
+
+def serialize_vertex(vertex: mgp.Vertex) -> Dict[str, Any]:
+    """
+    Serializes vertex to specified ElasticSearch schema.
+    Args:
+        vertex (mgp.Vertex): Reference to the vertex in Memgraph DB
+    Returns:
+        Dict[str, Any]: ElasticSearch object representation.
+    """
+    source = serialize_properties(vertex.properties.items())
+    source[LK_CATEGORIES] = [label.name for label in vertex.labels]
+    return {ACTION: {INDEX: {ID: str(vertex.id)}}, SOURCE: source}
+
+
+def serialize_edge(edge: mgp.Edge) -> Dict[str, Any]:
+    """
+    Serializes edge to specified ElasticSearch schema.
+    Args:
+        edge (mgp.Edge): Reference to the edge in Memgraph DB.
+    Returns:
+        Dict[str, Any]: ElasticSearch object representation.
+    """
+    source = serialize_properties(edge.properties.items())
+    source[LK_TYPE] = edge.type.name
+    return {ACTION: {INDEX: {ID: str(edge.id)}}, SOURCE: source}
```

Comment:
In my opinion not, because we need that to get JSON representation so dictionaries are very useful for this and also such a class would never be reused but we can discuss.

Reply:
@BorisTasevski ping here for discussion, I am not sure what is better also. But it seems this is the only comment left.

Reply:
I've expanded my initial comment

Reply:
Please address this